src/styles/index.scss

```diff
@@ -1,3 +1,68 @@
-body {
+$day-size: 100px;
+$gap: 1px;
+$days: (
+  mon 0,
+  tue 101px,
+  wed 202px,
+  thu 303px,
+  fri 404px,
+  sat 505px,
+  sun 606px
+);
+
+.body {
   margin: 0;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  width: ($day-size + $gap) * 7 - $gap;
+  gap: $gap;
+
+  @for $date from 1 through 31 {
+    & :nth-child(#{$date})::before {
+      font-family: Arial, sans-serif;
+      font-size: 30px;
+      position: absolute;
+      content: '#{$date}';
+      top: 50%;
+      left: 50%;
+      transform: translate(-50%, -50%);
```

Comment:
No need to duplicate the same styles (except `content`) 31 time
Define all common styles in `&__day::before`

BTW it's easier to center text using flex (for &__day). Also, font styles can be just inherited in ::before if you define them for &__day